### DISCUSSION TITLE
Whitespace not cleaned up when using arbitrary value [Prettier Plugin]

### DISCUSSION THREAD

**Discussion by faseehahmed1** [tailwindlabs/tailwindcss#13839] [Help]
`<div className="grid-cols-[2fr,5fr]>` //This works ✅
`<div className="grid-cols-[2fr, 5fr]>` //This does not work due to blank space before 5fr ❌

Can the prettier plugin not remove spaces "within" an arbitrary value?

**Comment by adamwathan** [comment]
Hey! This isn't something we do right now but maybe something we can think about. The challenge is that what you have there is actually just two separate classes because of the space:

```
grid-cols-[2fr,
5fr]
```

...and you can totally define valid CSS classes with those names:

```css
.grid-cols-\[2fr\, {
  /* ... */
}

.\35 fr {
  /* ... */
}
```

...so right now the plugin just sees those as two custom classes and sorts them to the beginning of the list.

It's technically unsafe to "fix" this on the chance that someone actually does have custom classes defined that could be misinterpreted as an arbitrary value with a space, but at the same time that would be a pretty stupid thing to do so maybe we can clean it up with an option to disable or something. Will talk about it with the team!
